.github/workflows/publish.yml

```
@@ -0,0 +1,39 @@
+# This workflow will upload a Python Package using Twine when a release is created
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#publishing-to-package-registries
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Upload Python Package
+
+on:
+  release:
+    types:
+      - published
+
+permissions:
+  contents: read
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install build
+      - name: Build package
+        run: python -m build
+      - name: Publish package
+        uses: pypa/gh-action-pypi-publish@27b31702a0e7fc50959f5ad993c78deac1bdfc29
```

just curious why you're using the commit hash here instead of something like:

```suggestion
        uses: pypa/gh-action-pypi-publish@v1
```

---

It's what's provided when generating the action from Github itself. On the action's repo under Usage, they suggest either using the tagged version, or using the commit SHA. Since that's what was pre-generated, I figured I'd just use that.

---

Weird. I feel like this won't get updates automatically if they need to update the action, whereas the other version specifier does. You'll have to periodically come back to this and update the commit hash, i would think.

---

I think their rationale was security considerations, but I can't load the page it references. In any case, I'm fine coming back to it to update it if it becomes incompatible.